### ISSUE TITLE
TabView does not support hiding Tabs

### ISSUE THREAD

**Issue by Anubarak** [primefaces/primevue#1113]
Could you eventually create support for hiding tabs in the `TabView` I already searched for similar issues but those are all about dynamic tabs.

https://codesandbox.io/s/bitter-microservice-mk7xl?file=/src/App.vue

```vue
<template>
  <button @click="toggleTab">Toggle Tab</button>
  <pre>{{ showTab }}</pre>
  <TabView>
    <TabPanel header="Header I"> Content I </TabPanel>
    <TabPanel header="Header II"> Content II </TabPanel>
    <TabPanel v-if="showTab === true" header="Header III">
      Content III
    </TabPanel>
  </TabView>
</template>
<script>
export default {
  data: () => ({
    showTab: false,
  }),
  methods: {
    toggleTab() {
      this.showTab = !this.showTab;
    },
  },
};
</script>
```
creates an 
> TypeError
> child.children.forEach is not a function

If you remove the `v-if` everything works fine.

**Comment by voquanghoa** [comment]
You can use 

```vue
<template>
  <button @click="toggleTab">Toggle Tab</button>
  <TabView>
    <TabPanel v-for="tab in tabs" :header="tab" :key="tab"> 
      Content {{tab}}
    </TabPanel>
  </TabView>
</template>
<script>
export default {
  data: () => ({
    showTab: false,
    tabs: ["Tab 1", "Tab 2", "Tab 3"]
  }),
  methods: {
    toggleTab() {
      this.tabs = [...this.tabs, "Tab " + (this.tabs.length + 1)];
    },
  },
};
</script>
<style lang="scss">
@import "./App.scss";
</style>
```

**Comment by Anubarak** [comment]
But this would require to store the content in a variable somehow. 
What would be a clean way to make this work with dynamic content inside the tabs. Or in other words, how would you convert the following? (besides creating a custom component for each tab and load it via `component :is`)

```vue
<TabView>
    <TabPanel header="Header I">
        <div> test </div>
    </TabPanel>
    <TabPanel header="Header II">
        <InputText />
        <FooBar>
            <div>Test</div>
        </FooBar>
    </TabPanel>
    <TabPanel v-if="showTab === true" header="Header III">
        <SomeOtherComponent />
    </TabPanel>
</TabView>
```

**Comment by voquanghoa** [comment]
Hello,

I think the issue you gave here is a bug in the library, I wish I can help you more on this by fixing it. But I'm just like you, I'm a user of the library. I also reported some other issues like this one https://github.com/primefaces/primevue/issues/1151 and I hope the team can fix them soon. But the developers are not very active, I think base on the status of the existing issue list.

Something I can suggest you in the meanwhile waiting for the team can fix the issue:

Use an enum to describe the tab content and check them 

```
<TabPanel v-for="tab in tabs" :header="tab.title" :key="tab.key"> 

      <template v-if="tab.type === TabContent.listProduct">
         <ListProductComponet .... />
     </template>

    <template v-if="tab.type === TabContent.listUser">
         <ListUserComponet .... />
     </template>
      
    <template v-if="tab.type === TabContent.configuration">
         <ConfigurationComponet .... />
     </template>
    ......
</TabPanel>
```

Note that Vue doesn't support switch-case (like Angular ng-switch https://angular.io/api/common/NgSwitch ) but I found this one https://github.com/vuejs/vue/issues/8097#issuecomment-486018518 can be used alternatively
